### PR TITLE
Add vendor app configurability

### PR DESCRIPTION
**Pull request by rondale-sc**
The goal of this PR is to allow the inlining of only the `vendor.css` file by passing a `vendorOnly` flag in the config for this addon.  While preserving the original behavior for both Glimmer and Ember applications.

The fundamental assumption of the code prior to this PR was that you'd always want to inline both `vendor.css` and `app.css`.  Separating these concerns is what necessitated the many changes introduced here.  As well as adding a testing harness.

### CODE REVIEW

**Comment by rondale-sc** [comment]
@mixonic I think this is ready for review

**Comment by mixonic** [comment]
@rondale-sc can you document what the intended result of this code is in the description?

**Comment by rondale-sc** [comment]
@mixonic Sure thing :+1: 

Added.  🍻

**Review by mixonic — CHANGES_REQUESTED** [review]
Can you add a test for `vendorOnly`? It also seems preferable if there is a way to specify `vendorOnly` without coupling to the name vendor. An API such as passing the files to inlined or a regex to filter them with seems preferable.

File: tests/index-test.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+const oneLineTrim = require('common-tags').oneLineTrim;
+const stripIndent = require('common-tags').stripIndent;
+
+const CSSReader = require('../index.js').CSSReader;
+
+const addon = require('../index.js');
+
+const test = QUnit.test;
+
+QUnit.module('CSSReader', function(hooks) {
```
Comment (mixonic) [inline tests/index-test.js]: I think this module name is wrong

File: tests/lib/css_injector-test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    input = yield createTempDir();
+    output = yield createTempDir();
+  }));
+
+  hooks.afterEach(co.wrap(function* () {
+    yield input.dispose();
+    yield output.dispose();
+  }));
+
+  test('when vendorOnly is specified only concats vendor.css to index.html', function(assert) {
```
Comment (mixonic) [inline tests/lib/css_injector-test.js]: Title is wrong, no vendorOnly

File: tests/index-test.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+  hooks.beforeEach(co.wrap(function* () {
+    input = yield createTempDir();
+  }));
+
+  hooks.afterEach(co.wrap(function* () {
+    yield input.dispose();
+  }));
+
+  test('concats vendor.css / app.css into style tag inside index.html', co.wrap(function* (assert) {
```
Comment (mixonic) [inline tests/index-test.js:27]: I think CSSReader tests are in the other module? My impression was this module was for full integration testing.

File: tests/index-test.js [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    });
+
+    let output = yield buildOutput(instance.postprocessTree('all', inputPath));
+
+    assert.deepEqual(output.read()['index.html'], oneLineTrim`
+      <style>a { background-color: blue; }</style>
+    `)
+  }));
+
+  test('postprocessTree vendorOnly', co.wrap(function* (assert) {
```
Comment (mixonic) [inline tests/index-test.js]: This is still testing vendorOnly

**Review by mixonic — CHANGES_REQUESTED** [review]
@rondale-sc perhaps you are WIP still, I noted down a few things here.

**Review by mixonic — APPROVED** [review]

**Comment by mixonic** [comment]
This looks excellent! Thank you @rondale-sc

**Pull request merged**